CandidateReranking.md

```diff
@@ -0,0 +1,368 @@
+| filters | [Filters[]](#Filter) | **Optional**: Array of filters used in the original query to search.
+
+#### ParsedInput
+| Param    | Type | Description
+|----------|------|--------|
+| input_to_extracted_keywords | Map<String, String> | **Optional**: Words extracted from `search_keywords` as the **Key** and their relavtive keywords identified by SemanticSearchAPI as the **Value**.
+| parsed_entities | ParsedEntity | **Optional**: Entities parsed from `search_keywords`.
+
+#### ParsedEntity
+| Param    | Type | Description
+|----------|------|--------|
+| entity_name | String | **Optional**: Name of the entity.
+| is_phrase | Boolean | **Optional**: Flags the entity as a phrase or not.
+| is_selected | Boolean | **Optional**: Flags the entity as sleceted or not.
+
+#### Keyword
+Entites returned from the SemanticSearchAPI.
+
+| Param    | Type | Description
+|----------|------|--------|
+| name | String | **Optional**: Name of the keyword, e.g. "Java".
+| type | String | **Optional**: Type of the keyword, e.g. "skill".
+| weight | Integer | **Optional**: Weight of the keyword as an int. \[0-100\]
+| relationships | [Relationship](#relationship) | **Optional**: Array of relationship objects which enrich the keyword with additional entities.
+
+#### Relationship
+A `relationship` is composed of a **\<semantic enrichment type\>** and the associated set of entities which can be classified under that **\<semantic enrichment type\>**. Accepted values for the **\<semantic enrichment type\>** include: `job_titles`, `occupations`, `skills`, `job_level`, `text_kernel_related_keywords`, `interesting_keywords`, `related_keywords`, `related_keywords_recruiter`, `related_keywords_jobseeker`, `raw_job_titles_jobseeker`, `raw_job_titles_recruiter`, `extracted_kewords`, `custom_keywords`, `related_search_terms`, and `job_titles_recruiter`.
+
+| Param    | Type | Description
+|----------|------|--------|
+| **\<semantic enrichment type\>** | [Entity[]](#entity) | **Optional**: Relationship to the keyword `name` value.
+
+#### Entity
+An `entity` is composed of a `name`, `weight`, and `selected` boolean. An `entity` enriches the original keyword with terms or phrases which share a close association. For example, if you were to have a `keyword` java, an enrichment entity might be java or programming.
+
+| Param    | Type | Description
+|----------|------|--------|
+| name | String | **Optional**: Name of enrichment `entity`. Ex., javascript, java, programming or software engineer.
+| weight | double | **Optional**: Weight of enriched term according to the Semantic Search API.
+| selected | boolean | **Optional**: Specific for the `related_search_terms` Semantic Enrichment. If selected the value is sent to the Reranker as a feature to be extracted for reranking else it is left out of reranking. All entites default to true.
```

Comment:
Not really a request for change, I'm just curious why would this ever be false? This effectively means that the front end is sending us an entity we will ignore?

Reply:
Indeed, they are sending us an entity we ignore if it is sent as false.